source_gen/test/generator_for_annotation_test.dart

```diff
@@ -142,12 +158,54 @@ $dartFormatWidth
     );
   });
 
+  test('applies to annotated directives', () async {
+    final builder = LibraryBuilder(
+      _StubGenerator<Deprecated>(
+        'Deprecated',
+        directiveBehavior: (element) => '// ${element.runtimeType}',
+        elementBehavior: (element) => '// ${element.runtimeType}',
+      ),
+    );
+    await testBuilder(
+      builder,
+      {
+        'a|lib/imported.dart': '',
+        'a|lib/part.dart': 'part of \'file.dart\';',
+        'a|lib/file.dart': '''
+      library;
```

Comment:
Are library statements also directives? Some generators also work based on library level annotations so we should validate this works as well.

Reply:
From the point of the element model - no.
Annotations on `library` are annotations on `LibraryElement`.
There is `applies to annotated libraries` test above.

Reply:
ack